test/docs/dash-test-workflow-saithrif.md

```diff
@@ -6,3 +6,64 @@ This document describes the DASH test workflow with SAI thrift.
+ * SAI header files are fed into a saithrift generator toolchain. This is a collection of Perl scripts which generates both client and server [Thrift](https://thrift.apache.org/) code. Thrift is a Remote Procedure Call (RPC) framework which uses an Interface Description language (IDL) schema to specify a client-server communications mechanism.
+* The saithrift toolchain yields Python client libraries which are imported and used by a test program to make remote SAI calls [**NOTE - diagram missing a label for thrift client lib imported into test script**]; and a server skeleton with RPC callback methods. These callbacks implement all the SAI RPC function signatures (CRUD operations upon all the conceptual SAI tables). These callbacks in turn make function calls into the vendor-supplied SAI library entrypoints, which must be implemented for each target (DUT).
+* Each DASH device vendor must implement a `libsai` shared library, i.e. a compiled `.so` module which can be linked to the server skeleton to produce a running SAI server image. The library has function calls supporting CRUD operations upon each DASH SAI header definition and attribute. The vendor `libsai` function entrypoints translate the abstract SAI CRUD operations into the underlying platform operations, e.g. typically via the vendor's Software Development Kit (SDK) libraries or equivalent. 
+* The vendor-supplied `libsai` is compiled with the saithrift server skeleton to produce an executable binary file which is a server daemon. When run, this daemon exposes a socket-based API endpoint. This endpoint is the SAI-thrift API server.
+Test programs can perform SAI RPC calls agains the saithrift endpoint. In the figure above, the PyTest test scripts make SAI calls to configure and query the DUT dataplane.
+### Physical or Simulated DUT
+A DUT may be a real hardware device (SmartNIC/DPU/IPU), or a software "simulation" thereof, for example a [P4-DPDK](https://github.com/p4lang/p4-dpdk-target) soft switch, or a [bmv2](https://github.com/p4lang/behavioral-model) model. Simulations will be used as reference targets to cross-check conformance tests and verify DASH SAI APIs. Performance of SW targets are not expected to meet DASH requirements, but nonetheless they serve as useful tools.
+### OTG/snappi Traffic Generator
+We test the DUT by configuring it as explained above; then sending it packets into a "front-panel" interace(s); then capturing and analyzing what the device sends back out the front panel interface(s). A Traffic Generator/Analyzer is used to emulate the behavior of many (even thousands) of hosts/servers in the normal datacenter environment. It can generate traffic streams simulating live traffic, capture what comes back, and analyze the results. This is the gist of DASH traffic tests.
```

Comment:
```suggestion
The DUT is tested by configuring it as explained before. Then sending packets to it into a "front-panel" interface(s); then capturing and analyzing what the device sends back out the front panel interface(s). A Traffic Generator/Analyzer is used to emulate the behavior of many (even thousands) of hosts/servers in the normal datacenter environment. It can generate traffic streams simulating live traffic, capture what comes back, and analyze the results. **This is the main idea of DASH traffic tests**.
```